api-server/src/server/boot/donate.js

```diff
@@ -42,117 +46,84 @@ export default function donateBoot(app, done) {
 
-    const fccUser = user
-      ? Promise.resolve(user)
-      : new Promise((resolve, reject) =>
-          User.findOrCreate(
-            { where: { email } },
-            { email },
-            (err, instance) => {
-              if (err) {
-                return reject(err);
-              }
-              return resolve(instance);
-            }
-          )
-        );
+    try {
+      const stripeCustomer = await stripe.customers.create({
+        email,
+        name
+      });
 
-    let donatingUser = {};
-    let donation = {
-      email,
-      amount,
-      duration,
-      provider: 'stripe',
-      startDate: new Date(Date.now()).toISOString()
-    };
-
-    const createCustomer = async user => {
-      let customer;
-      donatingUser = user;
-      try {
-        customer = await stripe.customers.create({
-          email,
-          card: id,
-          name
-        });
-      } catch (err) {
-        throw new Error('Error creating stripe customer');
-      }
-      log(`Stripe customer with id ${customer.id} created`);
-      return customer;
-    };
-
-    const createSubscription = async customer => {
-      donation.customerId = customer.id;
-      let sub;
-      try {
-        sub = await stripe.subscriptions.create({
-          customer: customer.id,
-          items: [
-            {
-              plan: `${donationSubscriptionConfig.duration[
-                duration
-              ].toLowerCase()}-donation-${amount}`
-            }
-          ]
-        });
-      } catch (err) {
-        throw new Error('Error creating stripe subscription');
-      }
-      return sub;
-    };
-
-    const createAsyncUserDonation = () => {
-      donatingUser
-        .createDonation(donation)
-        .toPromise()
-        .catch(err => {
-          throw new Error(err);
-        });
-    };
-
-    return Promise.resolve(fccUser)
-      .then(nonDonatingUser => {
-        // the logic is removed since users can donate without an account
-        return nonDonatingUser;
-      })
-      .then(createCustomer)
-      .then(customer => {
-        return createSubscription(customer).then(subscription => {
-          log(`Stripe subscription with id ${subscription.id} created`);
-          donation.subscriptionId = subscription.id;
-          return res.send(subscription);
-        });
-      })
-      .then(createAsyncUserDonation)
-      .catch(err => {
-        if (
-          err.type === 'StripeCardError' ||
-          err.type === 'AlreadyDonatingError'
-        ) {
-          return res.status(402).send({ error: err.message });
-        }
-        return res
-          .status(500)
-          .send({ error: 'Donation failed due to a server error.' });
+      const stripeSubscription = await stripe.subscriptions.create({
+        customer: stripeCustomer.id,
+        items: [
+          {
+            plan: `${donationSubscriptionConfig.duration[
+              duration
+            ].toLowerCase()}-donation-${amount}`
```

Comment:
```suggestion
            ]}-donation-${amount}`
```
Bit of a code smell. It implies to me that this might be doing two things 1) acting as a key and 2) being displayed to humans.

As far as I can see it's just used as a key, which is good, but means it can be lower case everywhere.

Reply:
TBH, we only offer month. So we could skip all {donationSubscriptionConfig.duration[duration].toLowerCase()
I might even drop the plan id for the unique product ids. We could pull those from the amount. 
But that is another PR.

Reply:
yes, there is another route that uses the same method. I refactor the month out completely in a separate pr.